### PR TITLE
feat: added "responseTo" field in worker emitted events

### PR DESCRIPTION
**Pull request by DuckBoss**
Currently uses cherry-picked commits from dependent PR: #140 which needs to be removed when the dependent PR gets approved and merged.

### CODE REVIEW

File: dbus/com.redhat.Yggdrasil1.xml [hunk 1]
```diff
@@ -62,6 +63,7 @@
             <arg type="s" name="worker" />
             <arg type="u" name="name" />
             <arg type="s" name="message_id" />
+            <arg type="s" name="response_to" />
```
Comment (subpop) [inline dbus/com.redhat.Yggdrasil1.xml:66]: We can avoid breaking *this* API by putting the `response_to` field below `message`.

**Review by subpop — CHANGES_REQUESTED** [review]
This should be able to be streamlined if in #140, we replace `message string` with `data map[string]string` everywhere, even in the `WorkerEvent` struct.

File: worker/echo/main.go [hunk 2]
```diff
@@ -31,9 +31,15 @@ func echo(
...
 ) error {
+	// Set "response_to" according to the rcvId of the message we received
+	echoResponseTo := rcvId
+	// Create new echoId for the message we are going to send
+	echoId := uuid.New().String()
+
 	if err := w.EmitEvent(
 		ipc.WorkerEventNameWorking,
-		rcvId,
+		echoId,
```
Comment (subpop) [inline worker/echo/main.go:41]: This is interesting. Prior to this change, each event emitted by the worker
would have the message ID of the received message in the first field, resulting
in output that looks like:

```
2023/09/15 12:52:45 echo: 53a8d7d5-e5cb-495c-9d18-2e055d566638: BEGIN: {}
2023/09/15 12:52:45 echo: 53a8d7d5-e5cb-495c-9d18-2e055d566638: WORKING: {"message":"echoing [104 101 108 108 111]"}
2023/09/15 12:52:45 echo: 53a8d7d5-e5cb-495c-9d18-2e055d566638: END: {}
```

Now however, the BEGIN and END events have the originating message ID first:

```
2023/09/15 12:52:45 echo: 53a8d7d5-e5cb-495c-9d18-2e055d566638: BEGIN: : {}
2023/09/15 12:52:45 echo: 80a83707-f982-4e82-82e5-90f06198f70e: WORKING: 53a8d7d5-e5cb-495c-9d18-2e055d566638: {"message":"echoing [104 101 108 108 111]"}
2023/09/15 12:52:45 echo: 53a8d7d5-e5cb-495c-9d18-2e055d566638: END: : {}
```

The two events handled by the `worker` package don't have any awareness to the
"response_to" field, so they don't print anything for that value. The WORKING
event is emitted by the echo worker, and it has created a new UUID for a
response message and puts *that* into the value of WORKING.

Overall, I don't think this is a significant flaw in this PR. But the way its
handled in this echo worker feels conflicted. I think it should continue to emit
the WORKING event, using the message ID of the received message as the first
field to the EmitEvent function. The "responseTo" function parameter has not
changed its meaning; it is still the value of the "response_to" field in the
received message.

**Review by subpop — CHANGES_REQUESTED** [review]

File: worker/echo/main.go (hunk 2 above)
Comment (DuckBoss) [reply worker/echo/main.go]: Thanks for the feedback!
I've reverted the commit that made this change so now it should continue to emit the message ID of the received message in the EmitEvent function for the **WORKING** event.

**Review by subpop — APPROVED** [review]
LGTM

**Review by subpop — APPROVED** [review]
LGTM

**Pull request merged**